src/sockets/tests/matchmakingQueue.test.ts

```diff
@@ -24,9 +24,11 @@ describe('MatchmakingQueue', () => {
     expect(matchFoundCallback).not.toHaveBeenCalled();
 
     matchmakingQueue.addUser(getDefaultQueueEntry('6'));
-    const expectUsernames = ['1', '2', '3', '4', '5', '6'];
-    expect(matchFoundCallback).toHaveBeenCalledWith(expectUsernames);
-    expect(matchmakingQueue.getNumInQueue()).toEqual(0);
+    setTimeout(() => {
+      const expectUsernames = ['1', '2', '3', '4', '5', '6'];
+      expect(matchFoundCallback).toHaveBeenCalledWith(expectUsernames);
+      expect(matchmakingQueue.getNumInQueue()).toEqual(0);
+    }, 10000);
```

Comment:
Let’s not add time delays into the test code itself.

There is a way for you to fast forward time in the test code. Look it up in jest :). 

https://jestjs.io/docs/timer-mocks

I’ve used it in a test file. You should be able to find examples.